server/src/http.rs

```diff
@@ -276,22 +276,28 @@ impl<Q: QueryExecutor + 'static> Service<Q> {
         let runtime = self.engine_runtimes.bg_runtime.clone();
 
         header::optional::<String>(consts::CATALOG_HEADER)
+            .and(header::optional::<String>(consts::SCHEMA_HEADER))
             .and(header::optional::<String>(consts::TENANT_HEADER))
-            .and_then(move |catalog: Option<_>, tenant: Option<_>| {
-                // Clone the captured variables
-                let default_catalog = default_catalog.clone();
-                let default_schema = default_schema.clone();
-                let runtime = runtime.clone();
-                async {
-                    RequestContext::builder()
-                        .catalog(catalog.unwrap_or(default_catalog))
-                        .tenant(tenant.unwrap_or(default_schema))
-                        .runtime(runtime)
-                        .build()
-                        .context(CreateContext)
-                        .map_err(reject::custom)
-                }
-            })
+            .and_then(
+                move |catalog: Option<_>, schema: Option<_>, tenant: Option<_>| {
+                    // Clone the captured variables
+                    let default_catalog = default_catalog.clone();
+                    let default_schema = default_schema.clone();
+                    let runtime = runtime.clone();
+                    // FIXME: for compatibility, we may use tenant as the schema if schema is
+                    // missing.
+                    let schema = schema.or(tenant).unwrap_or(default_schema);
```

Comment:
Should gRPC/MySQL service also have this compatible handling?

Reply:
gRPC service has already handled this. As for MYSQL service, it doesn't include this part.

Reply:
I prefer to add a common method to handle this compatibility, also we may need a flag to control this.

Reply:
Actually, the ways to extract the schema and tenant header value differs from each other for different service implementation. I guess such a common method can't be defined to satisfy the two service (http & gRPC).

Reply:
Then how about adding a flag to control this behavior?

Reply:
OK. I will add such flag.